### DISCUSSION TITLE
Camera Module v3 "normal" versus "NOIR"

### DISCUSSION THREAD

**Discussion by dalogue1** [monkeymademe/CamUI#62] [General]
Since both cameras are the same except the "Normal" camera has an IR filter over the lens, I guess they will both be identified the same.

![image](https://github.com/user-attachments/assets/ade6fc1e-33d5-430d-a1bf-26c6db4accbb)

![image](https://github.com/user-attachments/assets/c4a74980-d11f-440f-8a19-4f1c19d74cba)

Is there any way to identify whether a camera is the "normal" or "NOIR"? Both the "Camera Module v3" and "Camera Module v3 Wide Angle" have the attribute

            "got_noir": "Yes",

**Comment by monkeymademe** [comment]
Because the noir and the normal cameras share the same sensors its really up to the system to say if there are different and at least with the v1 noir that I have I can't see anything too tell them apart. 

That being said I don't have the v2 or v3 noir to test with and if you look at the model name for the v3 normal vs the v3 wide there is a and difference in the sensor model - 'imx708' and 'imx708_wide' so at the least for later models it might be possible I don't know

If you happen to have the noir could you try the following code in python (without the server running) and paste me the output then maybe we are in luck

from picamera2 import Picamera2
test =  Picamera2.global_camera_info()
print(test)

**Reply by dalogue1** [reply]
I will be getting the imx708 NOIR soon and will forward the info.  

I am also ordering a CCTV IR light to help it work: https://www.ebay.com/itm/335474847284

My plan is to run one camera server or the other not both at the same time. I will probably use a cron job to switch the camera is running based upon the time of day.  That way the YT video will continuously display the video night or day.

**Comment by monkeymademe** [comment]
You should see this 

>>> print(test)
[{'Model': 'ov5647', 'Location': 2, 'Rotation': 0, 'Id': '/base/axi/pcie@120000/rp1/i2c@88000/ov5647@36', 'Num': 0}, {'Model': 'imx708', 'Location': 2, 'Rotation': 180, 'Id': '/base/axi/pcie@120000/rp1/i2c@80000/imx708@1a', 'Num': 1}]

I am on the pi5 and Camera 'Num' 0 is the noir 'Model': 'ov5647'..... so unless this ID '/base/axi/pcie@120000/rp1/i2c@88000/ov5647@36' is unique or the model names are better in later versions we have to see

**Comment by monkeymademe** [comment]
Fundamentally lets say we can't 'tell' the difference between the normal and the noir cameras... If you connect them both to the pi5 they will show up as separate cameras... and can if you like share the same config but technically they don't need to (when the config saving is working).... After that trust me you will be able to tell which camera is noir and not. So lets say you use OBS to stream to youtube and have a timer running to switch camera... Just need to change from video_feed_0 to video_feed_1
